cmd/report.go

```diff
@@ -0,0 +1,139 @@
+
+#### Benchmark report for Github
+
+These report will be generated by comparing local benchmark results against ones from another benchmark run.
+The report will show performance differences between both runs.
+
+It is formatted as a Markdown Github comment to use as part of the CI results.
+`
+
+func setupReportsCommand() *cobraext.Command {
+	var reportTypeCmdActions []cobraext.CommandAction
+
+	cmd := &cobra.Command{
+		Use:   "report",
+		Short: "Generate reports",
+		Long:  reportLongDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Println("Generate reports")
+
+			if len(args) > 0 {
+				return fmt.Errorf("unsupported report type: %s", args[0])
+			}
+
+			return cobraext.ComposeCommandActions(cmd, args, reportTypeCmdActions...)
+		}}
+
+	cmd.PersistentFlags().BoolP(cobraext.FailOnMissingFlagName, "m", false, cobraext.FailOnMissingFlagDescription)
+	cmd.PersistentFlags().BoolP(cobraext.ReportFullFlagName, "", false, cobraext.ReportFullFlagDescription)
+	cmd.PersistentFlags().StringP(cobraext.ReportOutputFlagName, "", string(outputs.OutputFile), cobraext.ReportOutputFlagDescription)
+
+	// add benchmark report creation subcommand
+	cmd.AddCommand(getBenchReportCommand())
+
+	return cobraext.NewCommand(cmd, cobraext.ContextPackage)
+}
+
+func getBenchReportCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "benchmark",
+		Short: "Generate a benchmark report",
+		Long:  "Generate a benchmark report comparing local results against ones from another benchmark run.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Printf("Generate benchmark reports\n")
+
+			failOnMissing, err := cmd.Flags().GetBool(cobraext.FailOnMissingFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.FailOnMissingFlagName)
+			}
+
+			reportOutput, err := cmd.Flags().GetString(cobraext.ReportOutputFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ReportOutputFlagName)
+			}
+
+			isFull, err := cmd.Flags().GetBool(cobraext.ReportFullFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ReportFullFlagName)
+			}
+
+			srcPath, err := cmd.Flags().GetString(cobraext.BenchReportSrcPathFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.BenchReportSrcPathFlagName)
+			}
+
+			targetPath, err := cmd.Flags().GetString(cobraext.BenchReportTargetPathFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.BenchReportTargetPathFlagName)
+			}
+
+			threshold, err := cmd.Flags().GetFloat64(cobraext.BenchThresholdFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.BenchThresholdFlagName)
+			}
+
+			if _, err := os.Stat(srcPath); err != nil {
+				err = fmt.Errorf("stat file failed (path: %s): %w", srcPath, err)
```

Comment:
Should this error indicate that is referring to the source path? As well as the other to the target path.